src/tests/unit.rs

```diff
@@ -0,0 +1,34 @@
+use engine::Engine;
+
+#[test]
+fn test_unit() {
+    let mut engine = Engine::new();
+
+    if let Ok(result) = engine.eval::<()>("let x = (); x") {
+        assert_eq!(result, ());
+    } else {
+        assert!(false);
+    }
+}
+
+#[test]
+fn test_unit_eq() {
+    let mut engine = Engine::new();
+
+    if let Ok(result) = engine.eval::<bool>("let x = (); let y = (); x == y") {
+        assert!(result);
```

Comment:
This should simply be `assert_eq!(engine.eval::<..>(..), Ok(true))`. The same is true for all the other tests, actually.

Reply:
Fair point, it is just that the first tests were written that way and the new ones followed, I might refactor them later, when I have time. Sadly, I don't have much of it this month